docs/getting_started/scaleway.md

```diff
@@ -0,0 +1,59 @@
+# Getting Started with kops on Scaleway
+
+**WARNING**: Scaleway support on kOps is currently in **alpha**, which means that it is in the early stages of development and subject to change, please use with caution.
+
+## Requirements
+
+* [kops version >= 1.18 installed](../install.md)
+* [kubectl installed](../install.md)
+* [Setup your SSH key](https://www.scaleway.com/en/docs/configure-new-ssh-key/) : id_rsa and id_ed25519 keys are supported
+* [Scaleway credentials](https://www.scaleway.com/en/docs/generate-api-keys/) : you will need at least an access key, a secret key and a project ID.
+* [S3 bucket and its credentials](https://www.scaleway.com/en/docs/storage/object/quickstart/) : the bucket's credentials may differ from the one used for provisioning the resources needed by the cluster. If you use a Scaleway bucket, you will need to prefix the bucket's name with `scw://` in the `KOPS_STATE_STORE` environment variable. For more information about buckets, see [here](../state.md)
+
+### Optional
+* If you wish to use your own domain name for your cluster, you'll need a [Scaleway managed domain](https://www.scaleway.com/en/docs/network/domains-and-dns/quickstart). If not, you can always use the [gossip DNS](https://kops.sigs.k8s.io/gossip/) by naming your cluster with the suffix `.k8s.local`.
+
+## Environment Variables
+
+It is important to set the following [environment variables](https://github.com/scaleway/scaleway-sdk-go/blob/master/scw/README.md):
+```bash
+export SCW_ACCESS_KEY="my-access-key"
+export SCW_SECRET_KEY="my-secret-key"
+export SCW_DEFAULT_PROJECT_ID="my-project-id"
+export SCW_DEFAULT_REGION="fr-par"
+export SCW_DEFAULT_ZONE="fr-par-1"
+# Configure the bucket name to store kops state
+export KOPS_STATE_STORE=scw://<bucket-name> # where <bucket-name> is the name of the bucket you set earlier
+# Scaleway Object Storage is S3 compatible so we just override some S3 configurations to talk to our bucket
+export S3_REGION=fr-par                     # or another scaleway region providing Object Storage
+export S3_ENDPOINT=s3.$S3_REGION.scw.cloud  # define provider endpoint
+export S3_ACCESS_KEY_ID="my-access-key"     # where <access-key-id> is the Spaces API Access Key for your bucket
+export S3_SECRET_ACCESS_KEY="my-secret-key" # where <secret-key> is the Spaces API Secret Key for your bucket
```

Comment:
Can't the `scw:` code use `SCW_ACCESS_KEY`, etc?

Reply:
S3 credentials and Scaleway credentials can be different so I think we should keep the option to give S3 credentials, but I could make them optional (if no S3 credentials are given, use Scaleway credentials instead). Would it be better ?

Reply:
My naïve expectation would be for it to use Scaleway environment variables if configured, falling back to S3 environment variables otherwise. Then I would expect the documentation to prefer the Scaleway config file / environment variables. But there could be reasons to do it this way.

Putting credentials in environment variables is a security weakness, as that exposes them to other processes on the host.

Reply:
https://github.com/kubernetes/kops/pull/14642

Reply:
I opened #14787 that adds the option to try to get and use scaleway credentials if S3 credentials are not given. Reading your comment once again I noticed that it was not what you suggested, but I'm confused because it makes no sense to go further if it's the scaleway credentials that are missing since nothing else will work.